src/network/url_downloader.cpp

```diff
@@ -102,7 +102,24 @@ download(QNetworkAccessManager* manager, const Time& timeout, QUrl const& url, P
 
     if (reply->error() != QNetworkReply::NoError)
     {
-        const auto msg = download_timeout.isActive() ? reply->errorString().toStdString() : "Network timeout";
+        const auto error_code = reply->error();
+        const auto error_string = reply->errorString().toStdString();
+
+        // Log the original error message at debug level
+        mpl::log(mpl::Level::debug,
+                 category,
+                 fmt::format("Qt error {}: {}", static_cast<int>(error_code), error_string));
+
+        const auto msg = (error_code == QNetworkReply::TimeoutError) ? "Network timeout" : error_string;
+
+        if (error_code == QNetworkReply::TimeoutError)
```

Comment:
Combine the assignment of `msg` with this `if` clause? Otherwise, this could be combined with the nested `if` statement.